api/v1alpha1/conversion_utils.go

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) 2023 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package v1alpha1
+
+import (
+	"encoding/json"
+	"strings"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// ConversionAnnotationKey is the annotation key to ensure lossless
+// conversion of objects between API versions.
+const ConversionAnnotationKey = "vmoperator.vmware.com/conversion"
+
+func splitFieldsPath(input string) []string {
+	return strings.Split(input, ".")
+}
+
+// addConversionAnnotation appends the JSON marshalled data to Object as a special
```

Comment:
Is there a reason we're not cribbing closer from CAPI: https://github.com/kubernetes-sigs/cluster-api/blob/main/util/conversion/conversion.go#L101? Related, why not crib their `UnmarshalData()` instead of doing that in the type conversion, which I think means we'll end up with basically duplicated code once we do the VirtualMachine type.